ztp/policygenerator/policyGen/policyBuilder.go

```diff
@@ -94,6 +104,19 @@ func (pbuilder *PolicyBuilder) Build(policyGenTemp utils.PolicyGenTemplate) (map
 					if err != nil {
 						return policies, err
 					}
+					if sFile.RemediationAction != utils.UnsetStringValue {
+
+						if sFile.RemediationAction != policies[output].(utils.AcmPolicy).Spec.RemediationAction &&
+							policies[output].(utils.AcmPolicy).Spec.RemediationAction != policyGenTemp.Spec.RemediationAction {
+							//remediationAction already overridden by a different sourceFile
+							// and the remediationAction does not match
+							return policies, errors.New("remediationAction conflict for policyName " + sFile.PolicyName)
+						} else {
+							// sourceFile level remediationAction overrides default/PGT level
+							acmPolicy.Spec.RemediationAction = sFile.RemediationAction
+							acmPolicy.Spec.PolicyTemplates[0].ObjDef.Spec.RemediationAction = sFile.RemediationAction
+						}
+					}
```

Comment:
I am thinking can me make these into a function that be called in function createAcmPolicy() like the function BuildObjectTemplate for complianceType? I feel it's more clean and make code consitstent.